### PR TITLE
Replace Azure CI by Github Actions

### PR DESCRIPTION
**Pull request by tpayet**
- Add gh actions for cargo check using rust nightly
- Add readme about actions workflows
- Add basic Dockerfile
- Add action workflow for docker publish
- Change check action to test action

### CODE REVIEW

File: .github/workflows/nightly-test.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  check:
+    name: MeiliDB
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install latest nightly
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
```
Comment (Kerollmops) [inline .github/workflows/nightly-test.yml]: We now compile on stable, would it be possible to change that?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM rustlang/rust:nightly
```
Comment (Kerollmops) [inline Dockerfile]: Same, would it be possible to make it compile on stable?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+WORKDIR /meilisearch
+
+COPY . .
+
+RUN cargo build
+
+CMD cargo run
```
Comment (Kerollmops) [inline Dockerfile]: Wouldn't it be preferable to compile using `cargo build --release`, copy the `/target/release/meilidb-http` binary into _another_ sub image. This way we avoid compiling it every times it run and avoid keeping useless dependencies like `cargo`, `gcc`.

File: .github/workflows/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# GitHub actions workflow for MeiliDB
+
+> **Note:**
+> - The repository is currently compiling with Rust Nighlty, we aim to compile on Rust stable as soon as possible.
```
Comment (Kerollmops) [inline .github/workflows/README.md]: We compile with Rust stable now

File: Dockerfile (hunk 3 above)
Comment (tpayet) [reply Dockerfile]: yup ;)

File: .github/workflows/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+> - We do not use [cache](https://github.com/actions/cache) yet but we could use it to speed up CI
+
+## Workflow
+
+-  On each pull request, we are triggering `cargo test`.
```
Comment (Kerollmops) [inline .github/workflows/README.md]:
```suggestion
- On each pull request, we are triggering `cargo test`.
```

**Comment by Kerollmops** [comment]
You do not have to add the `MEILI_NO_ANALYTICS` env variable because analytics are only send when using `cargo run`, my bad!

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+# Run
+FROM    alpine:3.10
+
+RUN     apk update --quiet
+RUN     apk add libressl
+RUN     apk add build-base
+
+COPY    --from=compiler /meilisearch/target/release/meilidb-http .
+
+CMD     ./meilidb-http
```
Comment (Kerollmops) [inline Dockerfile]: Did you try to run that? I remember that alpine isn't shipped with a libc and that was a pain in the ass to make Rust work without it and use musl instead. I would prefer using Debian.

Comment (tpayet) [reply Dockerfile]: This was not about the docker image because we are not using the docker image for the test. It was the GitHub actions that was not using the latest rust version, and I just fixed it :)

Comment (Kerollmops) [reply Dockerfile]: Yes of course, that's just the Github Review interface that make it like those two comments are related :)

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+RUN     apk add build-base
+RUN     apk add libressl-dev
+
+RUN     curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+
+WORKDIR /meilisearch
+
+COPY    . .
+
+ENV     RUSTFLAGS="-C target-feature=-crt-static"
```
Comment (Kerollmops) [inline Dockerfile]: Why did you specified that line? To make it act like a fully static binary? Does it work?

File: README.md [hunk 8]
```diff
@@ -1,6 +1,7 @@
+
```
Comment (Kerollmops) [inline README.md]: Why did you add a new line here? :)

**Comment by Kerollmops** [comment]
Would it be possible to change the main readme to tell that we use stable rust now?

File: Dockerfile (hunk 7 above)
Comment (Kerollmops) [reply Dockerfile]: Ok, that make it compile in fact
